core/config/gear.ex

```diff
@@ -27,7 +27,10 @@ defmodule AntikytheraCore.Config.Gear do
       kv: Croma.Map,
       domains: CustomDomainList,
       log_level: Level,
-      alerts: HandlerConfigsMap
+      alerts: HandlerConfigsMap,
+      # can be used to store instance-specific information
+      # whose structure needs to be managed by administrative gears
+      internal_kv: {Croma.Map, default: %{}}
```

Comment:
IIUC, since this struct is used like `%__MODULE__{kv: %{}...}` in `default/0`, this `default: %{}` isn't used.
It would be better to add `internal_kv: %{}` to `default/0`

Reply:
This `default: %{}` is required when reading a config which is written in the current format.
Since a config in an existing file lack "internal_kv" field, `new/1` in line 61 will cause an error when reading the config.
`default/0` will be called only when a config file itself doesn't exist.

Reply:
Do you mean the following?
* If `read/1` is called for an existing gear, `read/1` returns `internal_kv: %{}` with the other keys because `default: %{}` in this croma struct is used.
* If `read/1` is called for a new gear, `read/1` returns the other keys without `internal_kv: %{}` because `default/0` is called.

Reply:
> * If `read/1` is called for an existing gear, `read/1` returns `internal_kv: %{}` with the other keys because `default: %{}` in this croma struct is used.

Yes, I mean it.

> * If `read/1` is called for a new gear, `read/1` returns the other keys without `internal_kv: %{}` because `default/0` is called.

No.
In this case, `internal_kv: %{}` is also included.
Croma struct implicitly do this.

Reply:
Understand!
Thank you for clarify.